include/exec/system_scheduler.hpp

```diff
@@ -0,0 +1,663 @@
+
+    size_t max_concurrency() const noexcept;
+
+  private:
+    __exec_system_context_interface* impl_ = nullptr;
+
+  };
+
+  class system_scheduler {
+  public:
+
+    // Pointer that we ref count?
+    system_scheduler(__exec_system_scheduler_interface* scheduler_interface) : scheduler_interface_(scheduler_interface) {}
+
+    bool operator==(const system_scheduler& rhs) const noexcept {
+      return scheduler_interface_->equals(rhs.scheduler_interface_);
+    }
+
+  private:
+    friend system_sender tag_invoke(
+      stdexec::schedule_t, const system_scheduler&) noexcept;
+
+    friend stdexec::forward_progress_guarantee tag_invoke(
+      stdexec::get_forward_progress_guarantee_t,
+      const system_scheduler&) noexcept;
+
+    template <stdexec::sender S, std::integral Shape, class Fn>
+    friend system_bulk_sender<S, Shape, Fn> tag_invoke(       //
+      stdexec::bulk_t,                                        //
+      const system_scheduler& sch,                            //
+      S&& sndr,                                               //
+      Shape shape,                                            //
+      Fn fun)                                                 //
+      noexcept;
+
+    __exec_system_scheduler_interface* scheduler_interface_;
+    friend class system_context;
+  };
+
+  class system_sender {
+  public:
+    using is_sender = void;
+    using completion_signatures =
+      stdexec::completion_signatures< stdexec::set_value_t(), stdexec::set_stopped_t() >;
+
+    system_sender(__exec_system_sender_interface* sender_impl) : sender_impl_{sender_impl} {}
+
+  private:
+    template <class S, class R_>
+    struct __op {
+      using R = R_;
+
+      template<class F>
+      __op(system_sender&& snd, R&& recv, F&& initFunc) :
+          snd_{std::move(snd)}, recv_{std::move(recv)}, os_{initFunc(*this)} {
+      }
+      __op(const __op&) = delete;
+      __op(__op&&) = delete;
+      __op& operator= (const __op&) = delete;
+      __op& operator= (__op&&) = delete;
+
+      friend void tag_invoke(stdexec::start_t, __op& op) noexcept {
+        if(auto os = op.os_) {
+          os->start();
+        }
+      }
+
+      S snd_;
+      R recv_;
+      __exec_system_operation_state_interface* os_ = nullptr;
+    };
+
+    template <class R>
+    friend auto tag_invoke(stdexec::connect_t, system_sender&& snd, R&& rec) //
+      noexcept(std::is_nothrow_constructible_v<std::remove_cvref_t<R>, R>)
+        -> __op<system_sender, std::remove_cvref_t<R>> {
+
+      return __op<system_sender, std::remove_cvref_t<R>>{
+        std::move(snd),
+        std::move(rec),
+        [](auto& op){
+          __exec_system_receiver receiver_impl{
+            &op.recv_,
+            [](void* cpp_recv){
+              stdexec::set_value(std::move(*static_cast<R*>(cpp_recv)));
+            },
+            [](void* cpp_recv){
+              stdexec::set_stopped(std::move(*static_cast<R*>(cpp_recv)));
+            }};
```

Comment:
`set_error` is not promised by the paper. There was an error here and I removed this case. It seems that I should have removed other occurrences of `set_error`; will remove them now.

Reply:
I forget the deal with set_error in the paper. I think I just didn't have a good reason to put it here, but the expectation would be that a receiver would be capable of receiving errors. Whether we want to define some specific potential failure types or not I don't know.

Reply:
After thinking some more on this, I think we need to have `set_error` here. Otherwise, how would the the implementation trigger some runtime errors (lack of memory, failure to allocate resources, etc.)

@ericniebler : at some point, long long time ago, there was a push to remove requirements for needing `set_error`. I take it that we do not want in P2300 to remove `std_error` as a requirement for any receiver, right?

Reply:
there is no longer a requirement for a receiver to have a set_error.